guardian/newsletters
# create local preview file from csv

## What does this change?

Currently, the only way to test changes to the `EmailNewsletter` model and the `rowToNewsletter` function (which parses an array of strings to an `EmailNewsletter`) is to use the `yarn dev` command, which fetches the data from the CODE google sheet and uploads the result to the "DEVELOPMENT" s3 bucket. This creates the following problems:
 - multiple developers cannot simultaneously test different sets of changes to the data, as they would all have to use the CODE google sheet as an input and over-writing each other's output
 - testing a 'breaking change' on the  CODE sheet (necessary to test breaking changes to the model) would break the CODE/Development version of any other application that used the CODE API
 - unnecessary calls are made to S3 and google API  for what should be local testing

This PR adds a feature that creates a local JSON file of the newsletter output using either a local csv file or the CODE spreadsheet as a data source.

## How to test
 - check out the branch
 - run `yarn preview` and view the file at `preview/preview.json`
 - make a changes to `preview/sampleData.csv` and/or `src/jobs/newsletters.ts`
 - run `yarn preview` again  to see the change reflected in `preview/preview.json`

## How can we measure success?
Changes to the data model / spreadsheet format can be tested more easily and more thoroughly, without  impacting the CODE API output.

## Have we considered potential risks?
The preview functions and associated files are outside the ./src folder, so will not be included in the build.
